base_layer/core/src/mempool/mempool_storage.rs

```diff
@@ -166,32 +166,27 @@ where T: BlockchainBackend
             );
         }
 
-        let prev_tip_height = removed_blocks
-            .last()
-            .expect("Added empty set of blocks on reorg.")
-            .header
-            .height;
-        let new_tip_height = new_blocks
-            .last()
-            .expect("Removed empty set of blocks on reorg.")
-            .header
-            .height;
+        let previous_tip = removed_blocks.last().map(|block| block.header.height);
+        let new_tip = new_blocks.last().map(|block| block.header.height);
+
         self.insert_txs(
             self.reorg_pool
                 .remove_reorged_txs_and_discard_double_spends(removed_blocks, &new_blocks)?,
         )?;
         self.process_published_blocks(new_blocks)?;
 
-        if new_tip_height < prev_tip_height {
-            debug!(
-                target: LOG_TARGET,
-                "Checking for time locked transactions in unconfirmed pool as chain height was reduced from {} to {} \
-                 during reorg.",
-                prev_tip_height,
-                new_tip_height,
-            );
-            self.pending_pool
-                .insert_txs(self.unconfirmed_pool.remove_timelocked(new_tip_height))?;
+        if let (Some(previous_tip_height), Some(new_tip_height)) = (previous_tip, new_tip) {
+            if new_tip_height < previous_tip_height {
+                debug!(
+                    target: LOG_TARGET,
+                    "Checking for time locked transactions in unconfirmed pool as chain height was reduced from {} to \
+                     {} during reorg.",
+                    previous_tip_height,
+                    new_tip_height,
+                );
+                self.pending_pool
+                    .insert_txs(self.unconfirmed_pool.remove_timelocked(new_tip_height))?;
+            }
```

Comment:
```suggestion
            }
            else {
             info!(.... "Processed reorg, but no reorg was required")
            }
```

Reply:
This means we can see in the logs if this branch is ever hit. Could also be at `debug` level

Reply:
Good idea - will add a debug to indicate that the new tip was >= previous tip and so checking for time-locked in unconfirmed was not necessary